themes/bootstrap5/js/facets.js

```diff
@@ -116,18 +116,318 @@ VuFind.register('facetList', function FacetList() {
+    if (elem.classList.contains('exclude')) {
+      elem.classList.toggle('selected');
+    } else {
+      let facet;
+      if (elem.classList.contains('facet')) {
+        facet = elem;
+      } else {
+        facet = elem.closest('.facet');
+      }
+      if (!facet.parentElement.classList.contains('checkboxFilter')) {
+        facet.classList.toggle('active');
+      }
+
+      let icon = elem.closest('a').querySelector('.icon');
+      if (icon !== null) {
+        const newCheckedState = icon.dataset.checked === 'false';
+        let attrs = {};
+        attrs.class = 'icon-link__icon';
+        attrs['data-checked'] = (newCheckedState ? 'true' : 'false');
+        icon.outerHTML = VuFind.icon(newCheckedState ? 'facet-checked' : 'facet-unchecked', attrs);
+      }
+    }
+  }
+
+  function handleClickedFacet(e) {
+    e.preventDefault();
+    let elem = e.currentTarget;
+
+    // Switch data-multi-filters-modified to keep track of changed facets
+    const currentAttrVal = elem.getAttribute('data-multi-filters-modified');
+    const isOriginalState = currentAttrVal === null || currentAttrVal === 'false';
+    elem.setAttribute('data-multi-filters-modified', isOriginalState);
+    toggleSelectedFacetStyle(elem);
+  }
+
+  function multiFacetsSelectionToggle() {
+    isMultiFacetsSelectionActivated = this.checked;
+    const count = rangeSelectorIds.length;
+    for (let i = 0; i < count; i++) {
+      let form = document.querySelector('form#' + rangeSelectorIds[i]);
+      if (form !== null) {
+        form.querySelector('input[type="submit"]').classList.toggle('hidden');
+      }
+    }
+    let buttons = document.getElementsByClassName('apply-filters');
+    for (let i = 0; i < buttons.length; i++) {
+      buttons[i].classList.toggle('hidden');
+    }
+    let checkboxes = document.getElementsByClassName('js-user-selection-multi-filters');
+    for (let i = 0; i < checkboxes.length; i++) {
+      checkboxes[i].checked = isMultiFacetsSelectionActivated;
+    }
+    if (!isMultiFacetsSelectionActivated) {
+      const elems = document.querySelectorAll('[data-multi-filters-modified="true"]');
+      for (const elem of elems) {
+        elem.setAttribute('data-multi-filters-modified', "false");
+        toggleSelectedFacetStyle(elem);
+      }
+    }
+    const event = isMultiFacetsSelectionActivated ? activation_event : deactivation_event;
+    VuFind.emit(event);
+  }
+
+  function registerCallbackOnApply(callback) {
+    callbackOnApply = callback;
+  }
+
+  function registerCallbackWhenDeactivated(callback) {
+    callbackWhenDeactivated = callback;
+  }
+
+  function applyClickHandling(context) {
+    let finalContext = (typeof context === "undefined") ? defaultContext : context;
+    finalContext.classList.toggle('multi-facet-selection');
+    finalContext.querySelectorAll('a.facet:not(.narrow-toggle), .facet a').forEach(function addListeners(link) {
+      link.addEventListener('click', function handling(e) {
```

Comment:
While debugging I noticed that `handling` is not very descriptive name for this function. I'd suggest something like `handleMultiSelectionClick` instead.